integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableReader.java

```diff
@@ -291,7 +292,12 @@ private CarbonTableCacheModel getValidCacheBySchemaTableName(SchemaTableName sch
         // Use block distribution
         List<List<CarbonLocalInputSplit>> inputSplits = new ArrayList(
             result.stream().map(x -> (CarbonLocalInputSplit) x).collect(Collectors.groupingBy(
-                carbonInput -> carbonInput.getSegmentId().concat(carbonInput.getPath()))).values());
+                carbonInput -> {
+                  if (FileFormat.ROW_V1.equals(carbonInput.getFileFormat())){
+                    return carbonInput.getSegmentId().concat(carbonInput.getPath()).concat(carbonInput.getStart() + "");
```

Comment:
why it needs to concat 'carbonInput.getStart()' when file format is ROW_V1?

Reply:
Only for the map key，without 'carbonInput.getStart()'，and only 'carbonInput.getSegmentId().concat(carbonInput.getPath())',  the some small files would be regarded as the same file  and it would be ingore. 
And  I think the bug doesn't cause in COLUMN_V3,  the COLUMN_V3 status contains compacted and success, however it queryed normally, so I only changed the ROW_V1 format.